### PR TITLE
[GraphQl] Adding Wishlist GraphQl coverage

### LINKED ISSUE #28551: Customer :: MyAccount :: Wishlist Management
As a Magento End Customer, 
I would like the ability to create multiple wishlists and add products to it
So that
I can create a personalized collection of products and save them in MyAccount for future reference. 

**Acceptance Criteria**

Wishlist functionality is only available for logged in customers. 
Wishlist functionality is only available via graphql if enabled in Magento admin. 
Customers can create one or more wishlists. 
Customers can add/remove one or more products from the wishlist. 
Customers can copy/move products between wishlists. 
Each Wishlist must have a shareable link. 
Open source only supports a single wishlist. 
Commerce edition must support more than one wishlist. 

**Approved Schema**

- https://github.com/magento/architecture/blob/master/design-documents/graph-ql/coverage/Wishlist.graphqls

### PR DESCRIPTION
**Pull request by eduard13**
### Description (*)
This PR implements the Wishlist GraphQl

[Solution Architecture](https://github.com/magento/architecture/blob/master/design-documents/graph-ql/coverage/Wishlist.graphqls)

### Related Pull Requests
https://github.com/magento/partners-magento2ee/pull/247

### Fixed Issues (if relevant)

1. [Customer :: MyAccount :: Wishlist Management](https://github.com/magento/community-features/issues/258)

### Manual testing scenarios (*)

Please check the following [document](https://github.com/magento/architecture/blob/master/design-documents/graph-ql/coverage/add-items-to-cart-single-mutation.md#options), in order to add different types of products to wishlist.

#### Add products to wishlist

```text
mutation {
  addProductsToWishlist(
    wishlistId: 0, // It may be always 0 for CE version
    wishlistItems: [
      {
        sku: "24-MB01"
        quantity: 3
      }
    ]
) {
    userInputErrors {
      code
      message
    }
    wishlist {
      id
      name
      sharing_code
      items_count
      items {
        id
        description
        qty
        added_at
      }
    }
  }
}
```

#### Update wishlist item
```text
mutation {
  updateProductsInWishlist(
    wishlistId: <wishlist_id>,
    wishlistItems: [
    {
  	wishlist_item_id: <wishlist_item_id>,
  	quantity: <quantity>,
        description: "Test Description"
    }
    ]
) {
    userInputErrors {
      code
      message
    }
    wishlist {
      id
      sharing_code
      items_count
      items {
        id
        description
        qty
        added_at
      }
    }
  }
}
```

#### Remove wishlist item
```text
mutation {
  removeProductsFromWishlist(
    wishlistId: <wishlist_id>, 
    wishlistItemsIds: [<wishlist_item_id>]
) {
    wishlist {
      name
      items_count
    }
  }
}
```

### Contribution checklist (*)
 - [ ] Pull request has a meaningful description of its purpose
 - [ ] All commits are accompanied by meaningful commit messages
 - [ ] All new or changed code is covered with unit/integration tests (if applicable)
 - [ ] All automated tests passed successfully (all builds are green)

### CODE REVIEW

**Comment by eduard13** [comment]
@magento run Static Tests

**Comment by eduard13** [comment]
@magento run Static Tests

**Comment by eduard13** [comment]
@magento run all tests

**Comment by eduard13** [comment]
@magento run all tests

File: app/code/Magento/WishlistGraphQl/Model/Resolver/AddProductsToWishlistResolver.php [hunk 1]
```diff
@@ -0,0 +1,129 @@
...
+use Magento\Wishlist\Model\Wishlist\Config as WishlistConfig;
+use Magento\Wishlist\Model\Wishlist\Data\Error;
+use Magento\Wishlist\Model\Wishlist\Data\WishlistItemFactory;
+use Magento\Wishlist\Model\WishlistFactory;
+use Magento\WishlistGraphQl\Mapper\WishlistDataMapper;
+
+/**
+ * Adding products to wishlist resolver
+ */
+class AddProductsToWishlistResolver implements ResolverInterface
```
Comment (rogyar) [inline app/code/Magento/WishlistGraphQl/Model/Resolver/AddProductsToWishlistResolver.php]: As a rule, we don't prefix resolvers with the "Resolver" keyword. We have some cases but they come from the past. It's not about something critical, but with the prefix we have quite a long class/file name. 
I would not put the prefix

Comment (eduard13) [reply app/code/Magento/WishlistGraphQl/Model/Resolver/AddProductsToWishlistResolver.php]: Was checking the resolvers from the same folder. However, the suffixes were removed.

**Comment by eduard13** [comment]
@magento run Static Tests, Magento Health Index

**Comment by eduard13** [comment]
@magento run Static Tests, Magento Health Index

**Comment by eduard13** [comment]
@magento run all tests

File: app/code/Magento/Wishlist/Model/Wishlist/AddProductsToWishlist.php [hunk 2]
```diff
@@ -0,0 +1,162 @@
...
+        return $wishlistOutput;
+    }
+
+    /**
+     * Add product item to wishlist
+     *
+     * @param Wishlist $wishlist
+     * @param WishlistItem $wishlistItem
+     */
+    private function addItemToWishlist(Wishlist $wishlist, WishlistItem $wishlistItem)
```
Comment (rogyar) [inline app/code/Magento/Wishlist/Model/Wishlist/AddProductsToWishlist.php]: Please, specify the return type

File: app/code/Magento/WishlistGraphQl/Model/Resolver/CustomerWishlistResolver.php [hunk 3]
```diff
@@ -42,6 +43,7 @@ public function resolve(
         array $value = null,
         array $args = null
     ) {
+        // Todo: Check if wishlist is enabled
```
Comment (rogyar) [inline app/code/Magento/WishlistGraphQl/Model/Resolver/CustomerWishlistResolver.php]: Please, do not forget about this todo

**Comment by eduard13** [comment]
@magento run Static Tests

**Comment by eduard13** [comment]
@magento run all tests

**Comment by eduard13** [comment]
@magento run all tests

**Comment by eduard13** [comment]
@magento run Database Compare

**Comment by eduard13** [comment]
@magento run all tests

**Comment by eduard13** [comment]
@magento run Static Tests

**Comment by eduard13** [comment]
@magento run all tests

**Comment by eduard13** [comment]
Hi @paliarush, could you please also adjust the approved wishlist schema, by replacing the arguments with camelcase format?
Thank you.

**Comment by eduard13** [comment]
@magento run all tests

**Review by rogyar — APPROVED** [review]

**Comment by magento-engcom-team** [comment]
Hi @rogyar, thank you for the review.
[ENGCOM-7685](https://jira.corp.magento.com/browse/ENGCOM-7685) has been created to process this Pull Request

**Comment by lenaorobei** [comment]
Please check failed web-api tests https://public-results-storage-prod.magento-testing-service.engineering/reports/magento/partners-magento2ee/pull/248/da6b0e91b08de451def0496ff26e663a/WebApi/allure-report-b2b-WebAPI-GRAPHQL/index.html

**Comment by eduard13** [comment]
@magento run WebAPI Tests

**Comment by eduard13** [comment]
Hi @lenaorobei those failing tests are caused by MultipleWishlist related PR. Please check this [comment](https://github.com/magento/partners-magento2ee/pull/248#issuecomment-633895470) for additional information

**Comment by lenaorobei** [comment]
In this case those PRs should be delivered as related and builds triggered for corresponding ce/ee branches.

**Comment by lenaorobei** [comment]
@magento run all tests

**Comment by eduard13** [comment]
@lenaorobei looks like everything is green ✅ here. Are we able to proceed further with this one or are there any additional requirements from my side?
Thank you.

**Comment by eduard13** [comment]
@magento run all tests

**Comment by lenaorobei** [comment]
@dthampy this PR is ready for QA.

**Comment by keharper** [comment]
I sent @eduard13 a schema file that contains updated descriptions.

**Comment by eduard13** [comment]
@magento run all tests

**Comment by eduard13** [comment]
Thank you @keharper, I've just updated it.

**Comment by eduard13** [comment]
@magento run all tests

**Comment by eduard13** [comment]
@magento run all tests

**Pull request merged**